src/Orleans.Runtime/Catalog/ActivationData.cs

```diff
@@ -387,6 +387,17 @@ public void ResetRunning(Message message)
         /// </summary>
         private List<Message> waiting;
 
+        private enum LoadState
+        {
+            Low,
+            Hot,
+            Overload
+        }
+
+        private LoadState loadState = LoadState.Low;
+        int lifetimeHighLoad = 0;
+        int spikeHighestLoad = 0;
```

Comment:
I like the idea of more monitoring, but this adds 3 new fields per activation. Can we do it in a cheaper way?